app/bundles/PageBundle/EventListener/BuildJsSubscriber.php

```diff
@@ -55,22 +55,29 @@ public static function getSubscribedEvents()
      */
     public function onBuildJs(BuildJsEvent $event)
     {
-        $pageTrackingUrl = str_replace(
-            ['http://', 'https://'],
-            '',
-            $this->router->generate('mautic_page_tracker', [], UrlGeneratorInterface::ABSOLUTE_URL)
-        );
+        $pageTrackingUrl = $this->router->generate('mautic_page_tracker', [], UrlGeneratorInterface::ABSOLUTE_URL);
+        // Determine if this is https
+        $parts           = parse_url($pageTrackingUrl);
+        $scheme          = $parts['scheme'];
+        $pageTrackingUrl = str_replace(['http://', 'https'], '', $pageTrackingUrl);
+
         $pageTrackingCORSUrl = str_replace(
-            ['http://', 'https://'],
```

Comment:
Doesn't this result in having a `$pageTrackingCORSUrl` that starts with `//` if their mautic is behind an SSL? Same for `$pageTrackingUrl` and `$contactIdUrl`. As an example: https://repl.it/FF78/0